.circleci/config.yml

```diff
@@ -1,35 +1,47 @@
 version: 2
 
+common-steps:
+  - &install-deps
+    run:
+      name: Install system-level and Python dependencies
+      command: |
+        apt-get update && apt-get install -y --no-install-recommends make python3 python3-venv python3-setuptools
+        python3 -m venv .venv
+        ./.venv/bin/pip install --require-hashes -r requirements/requirements.txt
+
 jobs:
   lint:
     docker:
-      - image: cimg/python:3.10.0
+      # We use a standard Debian image to mirror a typical developer environment.
+      # This should be updated whenever a new Debian stable version is available.
+      - image: debian:bullseye
     steps:
       - checkout
+      - *install-deps
       - run:
-          name: Install dependencies
-          command: pip install --require-hashes -r requirements/requirements.txt
-      - run:
-          name: Run linter
-          command: make docs-lint
+          name: Run lint
+          command: |
+            source .venv/bin/activate
+            make docs-lint
+
   linkcheck:
     docker:
-      - image: cimg/python:3.10.0
+      - image: debian:bullseye
     steps:
       - checkout
+      - *install-deps
       - run:
-          name: Install dependencies
-          command: pip install --require-hashes -r requirements/requirements.txt
-      - run:
-          name: Run link checker
-          command: make docs-linkcheck
-          no_output_timeout: 30m
```

Comment:
This timeout was increased in https://github.com/freedomofpress/securedrop-workstation-docs/pull/103 uniquely in this repo due to timeouts on nightly checks; I'm happy to keep an eye on things and we can re-introduce it if it's still warranted.

Reply:
Turns out it's indeed [failing again](https://app.circleci.com/pipelines/github/freedomofpress/securedrop-workstation-docs/696/workflows/b4b2d5d1-a66f-4360-abc1-859de5517217/jobs/680
) with `context deadline exceeded`, although it seems to run fine and quickly locally. :thinking:  Seems like it's hanging in CI on the first Whonix URL it encounters: https://www.whonix.org/wiki/Anon_Connection_Wizard

I'll try tweaking the Sphinx linkcheck settings before reintroducing the "no output" timeout in CI.